### PR TITLE
Remove YulStack::Machine enum.

### PR DESCRIPTION
**Pull request by aarlt**
**Depends on #13998**

### CODE REVIEW

File: test/cmdlineTests/strict_asm_options_in_non_asm_mode/err [hunk 1]
```diff
@@ -1 +1 @@
---yul-dialect and --machine are only valid in assembly mode.
+--yul-dialect only valid in assembly mode.
```
Comment (cameel) [inline test/cmdlineTests/strict_asm_options_in_non_asm_mode/err]:
```suggestion
--yul-dialect is only valid in assembly mode.
```

File: test/solc/CommandLineParser.cpp [hunk 2]
```diff
@@ -242,19 +242,19 @@ BOOST_AUTO_TEST_CASE(via_ir_options)
...
+	static vector<tuple<vector<string>, YulStack::Language>> const allowedCombinations = {
+		{{"--yul-dialect=evm", "--assemble"}, YulStack::Language::StrictAssembly},
+		{{"--yul-dialect=evm", "--yul"}, YulStack::Language::StrictAssembly},
+		{{"--yul-dialect=evm", "--strict-assembly"}, YulStack::Language::StrictAssembly},
+		{{"--assemble"}, YulStack::Language::Assembly},
+		{{"--yul"}, YulStack::Language::Yul},
+		{{"--strict-assembly"}, YulStack::Language::StrictAssembly},
+		{{"--assemble"}, YulStack::Language::Assembly},
+		{{"--yul"}, YulStack::Language::Yul},
+		{{"--strict-assembly"}, YulStack::Language::StrictAssembly},
```
Comment (cameel) [inline test/solc/CommandLineParser.cpp]: Without `--machine`, half of these are superfluous now.

Comment (aarlt) [reply test/solc/CommandLineParser.cpp]: true 😄

**Comment by cameel** [comment]
Since this is not mergeable without the base PR, please keep it as a draft for now and apply the `has dependencies` label.

File: libyul/YulStack.h [hunk 3]
```diff
@@ -60,14 +60,12 @@ struct MachineAssemblyObject
 };
 
 /*
- * Full assembly stack that can support EVM-assembly and Yul as input and EVM, EVM1.5 and
- * Ewasm as output.
+ * Full assembly stack that can support EVM-assembly and Yul as input and EVM.
```
Comment (nikola-matic) [inline libyul/YulStack.h]: There's another one of these on line 20.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `dbb57094f1b36d5d3f22ed82b228628225051fa7`:
```
Error: Trailing whitespace found:
Changelog.md:17: 

```
Please check that your changes are working as intended.

File: solc/CommandLineParser.cpp [hunk 4]
```diff
@@ -61,7 +61,6 @@ static string const g_strLicense = "license";
 static string const g_strLibraries = "libraries";
 static string const g_strLink = "link";
 static string const g_strLSP = "lsp";
-static string const g_strMachine = "machine";
```
Comment (ekpyron) [inline solc/CommandLineParser.cpp]: Maybe we should keep the command line option for now (accepting only , for backwards compatibility, but remove its docs?

I.e. maybe we should reduce the PR to just remove the enum in ``YulStack``.

We can also just make all of this an issue (and fully keep the enum for now, closing the PR), though - it's not like this saves us that much complication in general.

File: solc/CommandLineParser.cpp [hunk 5]
```diff
@@ -1208,15 +1207,12 @@ void CommandLineParser::processArgs()
...
 
 		if (m_args.count(g_strMachine))
 		{
 			string machine = m_args[g_strMachine].as<string>();
-			if (machine == g_strEVM)
-				m_options.assembly.targetMachine = Machine::EVM;
-			else
+			if (machine != g_strEVM)
 				solThrow(CommandLineValidationError, "Invalid option for --" + g_strMachine + ": " + machine);
 		}
```
Comment (cameel) [inline solc/CommandLineParser.cpp:1217]: I'd change the message here to say that `--machine` is deprecated and has no effect. And I'd also mention that it's deprecated in `--help`.

**Review by cameel — COMMENTED** [review]
Looks good apart from some missing adjustments to CLI help and messages.

By the way, I'm assuming it's reviewable new (since you requested my review)? It's still marked as draft.

**Pull request closed without merging**